romancal/patch_match/patch_match.py

```diff
@@ -0,0 +1,299 @@
+    coordinates in the cartesian system so that it avoids the distortions in plots
+    due to areas approaching the poles.
+    """
+    corners = (
+        (
+            patch["dec_corn1"],
+            patch["dec_corn2"],
+            patch["dec_corn3"],
+            patch["dec_corn4"],
+            patch["dec_corn1"],
+        ),
+        (
+            patch["ra_corn1"],
+            patch["ra_corn2"],
+            patch["ra_corn3"],
+            patch["ra_corn4"],
+            patch["ra_corn1"],
+        ),
+    )
+    corners = np.array(corners)
+    vec_corners = sgv.lonlat_to_vector(corners[0], corners[1])
+    return vec_corners
+
+
+def find_closest_tangent_point(patches, image_corners):
+    """
+    Out of all listed patches, find the closest tangent point to the center
+    coordinate of the image.
+    """
+    # To deal with the corner case, it is necessary to use spherical_geometery
+    # to average the the corners properly.
+    # Convert image corners to unit vectors.
+    vec_im_corners = image_coords_to_vec(image_corners)
+    im_center = np.array(normalize_vector(vec_im_corners.mean(axis=1)))
+    tangent_point_set = set()
+    patch_tangent_points = [
+        (patch["dec_projection_center"], patch["ra_projection_center"])
+        for patch in patches
+    ]
+    for tangent_point in patch_tangent_points:
+        tangent_point_set.add(tangent_point)
+    unique_tangent_points = list(tangent_point_set)
+    # Compute distance for each tangent point from im_center
+    dist = [
+        ((im_center - np.array(sgv.lonlat_to_vector(*tangent_point))) ** 2).sum()
+        for tangent_point in unique_tangent_points
+    ]
+    sorted_dist_indices = sorted(zip(dist, range(len(dist))))
+    sorted_tangent_points = [
+        unique_tangent_points[sorted_dist[1]] for sorted_dist in sorted_dist_indices
+    ]
+    closest_tangent_point = np.array(sgv.lonlat_to_vector(*sorted_tangent_points[0]))
+    # Now associate index of sorted_tangent_points with that of all patches
+    patch_tp_id = []
+    for patch_tp in patch_tangent_points:
+        for i, tp in enumerate(sorted_tangent_points):
+            if tp == patch_tp:
+                patch_tp_id.append(i)
+    return closest_tangent_point, patch_tp_id
+
+
+def normalize_vector(vec):
+    """
+    Normalize a 3d vector to have length 1. Only works on 1d arrays.
+    """
+    return vec / np.sqrt((vec**2).sum())
+
+
+def veccoords_to_tangent_plane(vertices, tangent_point_vec):
+    """
+    Convert the spherical geometry vectors to tangent plane coordinates
+    in arcseconds. This algorithm is not precise, but should be good
+    enough for now (and besides, the goal here is visualizaion, not
+    ultra-precision). This also breaks down numerically very near the
+    poles.
+    """
+    # First compute the tangent plane axis vectors.
+    x_axis = normalize_vector(np.cross([0, 0, 1], tangent_point_vec))
```

Comment:
consider using https://github.com/spacetelescope/spherical_geometry/blob/5de7fc79d8e3cb4f1c0a9338acf06b350d17b15b/spherical_geometry/great_circle_arc.py#L64-L79

it is likely more accurate due to quad math library